### PR TITLE
Added Census Tract Aggregated Micro-data from EPA Risk-Screening Environmental Indicators (RSEI) model

### PR DESCRIPTION
**Pull request by saran-ahluwalia**
Addresses #1070 

To replicate: 

1. ` poetry run python3 data_pipeline/application.py etl-run --dataset epa_rsei_aggregate`
2. `poetry run python3 data_pipeline/application.py score-run`
3. In the comparison tool Jupyter notebook run the entire notebook to generate scores for RSEI Aggregated Microdata based on the percentile rank of 75 for scoring comparison tool.
4. There is the use of binning values into general percentile ranges for easier stratification. The rationale is detailed below in the source code itself. In general, I attempt to answer: "where does this score fall in, relative to other scores"

### CODE REVIEW

File: data/data-pipeline/data_pipeline/score/field_names.py [hunk 1]
```diff
@@ -153,6 +153,40 @@
...
+
+EPA_RSEI_SCORE_PERCENTILE_LABEL_FIELD = (
+    "Percentile intervals for overall risk scores"
+)
+EPA_RSEI_CSCORE_PERCENTILE_LABEL_FIELD = (
+    "Percentile intervals for other risk scores (Cancer toxicity weights)"
+)
+EPA_RSEI_NCSCORE_PERCENTILE_LABEL_FIELD = (
+    "Percentile intervals for other risk scores (Noncancer toxicity weights)"
+)
```
Comment (saran-ahluwalia) [inline data/data-pipeline/data_pipeline/score/field_names.py:184]: These columns reference bins for percentiles. I am wondering if the word choice is appropriate?

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        # when performing more advanced statistical analyses of each component of the distribution. This is pertienent
+        # if we seek to combine this data with population characteristics and demographics - for example, ACS and CDC SVI Index -
+        # in order to chracterize each section of the "higher scoring" tracts for cancer-related and non-cancer related toxicity
+        self.df[field_names.EPA_RSEI_SCORE_PERCENTILE_LABEL_FIELD] = pd.cut(
+            self.df[field_names.EPA_RSEI_SCORE_OUTPUT_FIELD],
+            right=True,  # include right-most interval
+            bins=perc_func_overall_risk_score,
+            labels=labels_for_cut_function,
+            ordered=True,
+        )
```
Comment (saran-ahluwalia) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:199]: I wanted to discuss this with @lucasmbrown-usds and discuss some of the initial thought processes above.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+        if (
+            not self.df[self.GEOID_TRACT_FIELD_NAME]
+            .apply(lambda x: len(str(x)))
+            .eq(expected_census_tract_field_length)
+            .all()
+        ):
+            raise ValueError(
+                f"GEOID Tract must be length of {expected_census_tract_field_length}"
```
Comment (saran-ahluwalia) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:254]: We seem to reuse this same check in many data source ETL classes. Does it make sense to delegate this to the superclass?

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 2 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:204]: Another way to stratify would be something like this:

```
def binning_function(col, cut_points, labels = None): 

    #Define min and max values:
    minval = col.min()
    maxval = col.max()

    # create list by adding min and max to cut_points
    break_points = [minval] + cut_points + [maxval]


    # if no labels provided, use
    # default labels 0...(n - 1)
    if not labels:
        labels = range(len(cut_points) + 1)

    # Binning using cut
    # function from Pandas
    column_binning = pd.cut(col, bins = break_points, labels = labels, include_lowest = True)
    return column_binning

# Binning overall risk score:
cut_points = [90, 140, 190]# hypothetical example of values
labels = ["low", "medium", "high", "very high"]
self.df[field_names.EPA_RSEI_SCORE_LABEL_FIELD] = binning_function(self.df[field_names.EPA_RSEI_SCORE_OUPUT_FIELD], cut_points, labels)
```

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+            self.NUMBER_FACILITIES_INPUT_FIELD,
+            self.NUMBER_RELEASES_INPUT_FIELD,
+            self.NUMBER_CHEMICALS_INPUT_FIELD,
+            self.AVERAGE_TOXICITY_INPUT_FIELD,
+            self.SCORE_INPUT_FIELD,
+            self.POPULATION_INPUT_FIELD,
+            self.CSCORE_INPUT_FIELD,
+            self.NCSCORE_INPUT_FIELD,
+        ]
+        # The subsequent step make the current columns the first row
```
Comment (saran-ahluwalia) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:118]: Nit *steps

**Comment by saran-ahluwalia** [comment]
This reflects only aggregated micro-data for RSEI modeled data. However, it is worth pointing out that the Disaggregated Census Tract Microdata does afford us the ability to better characterize releases, chemical composition, the concentration of ay given chemical for release/media at grid cell (µg/m^3). This is certainly useful if one seeks to analyze state-level environmental indicators and screening tools (such as the Maryland EJSCREEN).

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+import pandas as pd
+
+from data_pipeline.etl.base import ExtractTransformLoad
+from data_pipeline.score import field_names
+from data_pipeline.utils import get_module_logger, unzip_file_from_url
+
+logger = get_module_logger(__name__)
+
+
+class EPARSEISCOREETL(ExtractTransformLoad):
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:12]: Same comment as on other PR: can we make this class name more readable? They're supposed to be CamelCase but the abbreviations throw it off.

Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:12]: Like `RiskScreeningEnvironmentalIndicatorsETL` or something. Could also rename the `epa_rsei_aggregated_microdata` references to `epa_risk_screening_environmental_indicators`.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+    """Class for 2019 Census Tract Aggregated micro-data
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:13]: nit: should probably reference RSEI in this tile, not just "census aggregated"

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+            field_names.EPA_RSEI_SCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_NCSCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_CSCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD,
+            field_names.EPA_RSEI_SCORE_THRESHOLD_FIELD,
+        ]
+
+        self.df: pd.DataFrame
+
+    def _custom_percentile(self, data, percentile):
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:60]: I'm not sure I understand why you needed to create this custom function and you didn't use something like `np.nanquantile(a= data, q= percentile)`?

File: data/data-pipeline/data_pipeline/score/field_names.py [hunk 8]
```diff
@@ -153,6 +153,40 @@
...
+
+EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD = "Percentile rank for overall risk score"
+
+EPA_RSEI_SCORE_THRESHOLD_FIELD = "At or above 90 for overall percentile ranking according to overall risk score"
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/score/field_names.py:188]: I think this field title is misleading compare to the .75 cutoff.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 3 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:254]: yea definitely!

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+                self.AVERAGE_TOXICITY_INPUT_FIELD: field_names.EPA_RSEI_AVERAGE_TOXICITY_OUTPUT_FIELD,
+                self.SCORE_INPUT_FIELD: field_names.EPA_RSEI_SCORE_OUTPUT_FIELD,
+                self.CSCORE_INPUT_FIELD: field_names.EPA_RSEI_CSCORE_OUTPUT_FIELD,
+                self.NCSCORE_INPUT_FIELD: field_names.EPA_RSEI_NCSCORE_OUTPUT_FIELD,
+                self.POPULATION_INPUT_FIELD: field_names.EPA_RSEI_POPULATION_OUTPUT_FIELD,
+            },
+            inplace=True,
+        )
+
+        # 5th, 25th, 50th, 75th and 99th percentiles for each score column
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:147]: How are you planning to use these bucketed "cut" columns? There's a lot of code used to create them, but we don't actually use them anywhere or produce any analysis based on them, so I'm wondering if making the code for this ETL class that much more complex is worth it if we never really use the results. Overall the class could be a lot cleaner if just output some of the scores.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+            ascending=True,
+            pct=True,
+        )
+
+        # This threshold was arbitrarily chosen.
+        # It would make sense to enrich this with facilities, industries, or chemical
+        # that would enable some additional form of sub-stratification when examining
+        # different percentile ranges that are derived above.
+        self.df[field_names.EPA_RSEI_SCORE_THRESHOLD_FIELD] = (
+            self.df[field_names.EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD] > 0.75
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:237]: Want to make .75 an attribute on the class (ie., in init define `self.EPA_RSEI_SCORE_THRESHOLD_CUTOFF = .75`) to make it easier to find?

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb [hunk 11]
```diff
@@ -314,6 +314,25 @@
...
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b39342aa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load Michigan EJSCREEN\n",
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:324]: wrong comment

File: data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb [hunk 12]
```diff
@@ -314,6 +314,25 @@
...
+    "epa_rsei_agggregate_data_path = (\n",
+    "    DATA_DIR / \"dataset\" / \"epa_rsei_aggregated_microdata\" / \"usa.csv\"\n",
+    ")\n",
+    "epa_rsei_agggregat_df = pd.read_csv(\n",
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/ipython/scoring_comparison.ipynb:328]: `epa_rsei_agggregat_df` -> `epa_rsei_agggregate_df`

File: data/data-pipeline/data_pipeline/score/field_names.py (hunk 1 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/score/field_names.py:184]: Yeah I'm not sure. See my comment above about whether these are necessary?

**Review by lucasmbrown-usds — CHANGES_REQUESTED** [review]

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 9 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:147]: The rationale is outline [here]( https://github.com/usds/justice40-tool/pull/1101/files#diff-b3ec2dbf735f5d657d3212c93e05e5bd3d6de42a4b0db7e7bcd20aa9f51e5829R185) and also in introductory comments [here](https://github.com/usds/justice40-tool/pull/1101#issue-1094744768)  (specifically #3 and #4).

I wanted to discuss this with you as stated [here](https://github.com/usds/justice40-tool/pull/1101/files#r779188783).

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 7 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:60]: Please see my comments below and in the PR introduction. I hope that is helpful.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 9 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:147]: Ok. The ratio of complexity to add these fields to the value they bring seems too high to me – I don't think it's worth it based on what I'm seeing it. We do not use these results anywhere that I can tell. 

Could you remove the code that creates these buckets for now? It'll be in our git history so we can always resuscitate them if necessary.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 7 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py]: Not pertinent anymore as there is no expressed desire to include stratification or examining different portions of the distribution of scores in conjunction with other datasets.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 5 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py]: How about `epa_rsei_aggregated_risk_environmental_indicator`? This ensures there is still specificity of the data. There is also disaggregated and other geographic units of measurements provided.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 3 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:158]: This is a good example of where `pandera` can be used to enforce a contract across all ETL classes simply by invoking a superclass's utility. I will be working with @sverchdotgov on this today.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 5 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py]: Understood - I will make the change.

File: data/data-pipeline/data_pipeline/etl/constants.py [hunk 13]
```diff
@@ -89,6 +89,11 @@
         "module_dir": "hud_recap",
         "class_name": "HudRecapETL",
     },
+    {
+        "name": "epa_rsei_aggregate",
+        "module_dir": "epa_rsei_aggregate",
+        "class_name": "EPARiskScreeningEnvironmentalIndicatorsETL",
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/constants.py:95]: These LGTM!

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+
+    def __init__(self):
+        self.AGGREGATED_RSEI_SCORE_FILE_URL = "http://abt-rsei.s3.amazonaws.com/microdata2019/census_agg/CensusMicroTracts2019_2019_aggregated.zip"
+
+        self.OUTPUT_PATH: Path = (
+            self.DATA_PATH
+            / "dataset"
+            / "epa_rsei_aggregated_risk_environmental_indicators"
+        )
+        self.EPA_RSEI_SCORE_THRESHOLD_CUTOFF = 0.75
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:30]: 🎉

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 4 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:94]: should still be "steps"?

Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:94]: This is a bit of an odd and complicated way to do what you're trying to achieve here. 

Why not just write, in the `extract` method: 

```
input_columns = [
            self.TRACT_INPUT_COLUMN_NAME,
            self.NUMBER_FACILITIES_INPUT_FIELD,
            self.NUMBER_RELEASES_INPUT_FIELD,
            self.NUMBER_CHEMICALS_INPUT_FIELD,
            self.AVERAGE_TOXICITY_INPUT_FIELD,
            self.SCORE_INPUT_FIELD,
            self.POPULATION_INPUT_FIELD,
            self.CSCORE_INPUT_FIELD,
            self.NCSCORE_INPUT_FIELD,
        ]
self.df = pd.read_csv(
            filepath_or_buffer=self.TMP_PATH
            / "epa_rsei_aggregated_risk_environmental_indicators"
            / "CensusMicroTracts2019_2019_aggregated.csv",
            # The following need to remain as strings for all of their digits, not get
            # converted to numbers.
            low_memory=False,
            names=input_columns,
        )
```

And then you don't need any of this complex combinations?

File: data/data-pipeline/data_pipeline/score/field_names.py [hunk 15]
```diff
@@ -153,6 +153,32 @@
     "Percent of individuals < 100% Federal Poverty Line in 2010"
 )
 
+# RSEI Aggregated Micro-data
+EPA_RSEI_NUMBER_FACILITIES_OUTPUT_FIELD = (
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/score/field_names.py:157]: We don't use "OUTPUT_" in any of the other names in this file. Can you just call them "_FIELD"? E.g., `EPA_RSEI_NUMBER_FACILITIES_FIELD`

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+            pct=True,
+        )
+
+        # This threshold was arbitrarily chosen.
+        # It would make sense to enrich this with facilities, industries, or chemical
+        # that would enable some additional form of sub-stratification when examining
+        # different percentile ranges that are derived above.
+        self.df[field_names.EPA_RSEI_SCORE_THRESHOLD_FIELD] = (
+            self.df[field_names.EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD]
+            > self.EPA_RSEI_SCORE_THRESHOLD_CUTOFF
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:141]: Nit: I think we use >= on most of these comparisons, not that it matters much in practice.

File: data/data-pipeline/data_pipeline/score/field_names.py [hunk 17]
```diff
@@ -153,6 +153,32 @@
...
+EPA_RSEI_CSCORE_OUTPUT_FIELD = "Sum of the risk-related scores of the cells in the tract (Cancer toxicity weights)"
+EPA_RSEI_NCSCORE_OUTPUT_FIELD = "Sum of the risk-related scores of the cells in the tract (Noncancer toxicity weights)"
+EPA_RSEI_POPULATION_OUTPUT_FIELD = (
+    "Sum of the population of the cells in the tract"
+)
+
+EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD = "Percentile rank for overall risk score"
+
+EPA_RSEI_SCORE_THRESHOLD_FIELD = (
+    "At or above 75 for overall percentile rank according to overall risk score"
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/score/field_names.py:179]: When this gets combined with all the other fields in the big `usa.csv`, "overall risk score" won't be descriptive enough for anyone to understand. Could you make this more descriptive? And please don't refer to percentile ranks, refer to percentiles.

File: data/data-pipeline/data_pipeline/score/field_names.py [hunk 18]
```diff
@@ -153,6 +153,32 @@
...
+)
+EPA_RSEI_NUMBER_RELEASES_OUTPUT_FIELD = "Number of releases affecting the tract"
+EPA_RSEI_NUMBER_CHEMICALS_OUTPUT_FIELD = (
+    "Number of chemicals affecting the tract"
+)
+EPA_RSEI_AVERAGE_TOXICITY_OUTPUT_FIELD = (
+    "Average toxicity-weighted concentration of the cells in the tract"
+)
+EPA_RSEI_SCORE_OUTPUT_FIELD = (
+    "Sum of the risk-related scores of the cells in the tract"
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/score/field_names.py:168]: Similar comment: "risk-related scores" is too vague. Please rename this field something like `"RSEI Risk Score"`.

File: data/data-pipeline/data_pipeline/score/field_names.py [hunk 19]
```diff
@@ -153,6 +153,32 @@
...
+)
+EPA_RSEI_CSCORE_OUTPUT_FIELD = "Sum of the risk-related scores of the cells in the tract (Cancer toxicity weights)"
+EPA_RSEI_NCSCORE_OUTPUT_FIELD = "Sum of the risk-related scores of the cells in the tract (Noncancer toxicity weights)"
+EPA_RSEI_POPULATION_OUTPUT_FIELD = (
+    "Sum of the population of the cells in the tract"
+)
+
+EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD = "Percentile rank for overall risk score"
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/score/field_names.py:176]: Please follow conventions of other fields, and don't create a new specific field for this, and instead use `field_names. EPA_RSEI_SCORE_OUTPUT_FIELD + field_names.PERCENTILE_FIELD_SUFFIX` to reference this. Thanks!

**Review by lucasmbrown-usds — CHANGES_REQUESTED** [review]
Requested a few changes, thanks!

File: data/data-pipeline/data_pipeline/score/field_names.py (hunk 18 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/score/field_names.py:168]: I am following the documentation from the EPA, but will make the change.

File: data/data-pipeline/data_pipeline/score/field_names.py (hunk 17 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/score/field_names.py:179]: Screen Shot 2022-01-13 at 1 17 58 PM

The RSEI score is a function of all of these [inputs](https://www.epa.gov/rsei/understanding-rsei-results). I can come up with something to convey a more meaningful interpretation.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 3 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:158]: This comment did not age well - as I said, as OS contributor I will continue to this project and will address this even after tomorrow. I care about this team's work.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 4 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:92]: @saran-ahluwalia , why did you mark this as resolved?

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+            inplace=True,
+        )
+
+        # Please note this: https://www.epa.gov/rsei/understanding-rsei-results#what
+        # Section: "What does a high RSEI Score mean?"
+        # This was created for the sole purpose to be used in the current
+        # iteration of Score L
+        self.df[
+            field_names.EPA_RSEI_SCORE_FIELD
+            + field_names.PERCENTILE_FIELD_SUFFIX
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:127]: 🎉

File: data/data-pipeline/data_pipeline/score/field_names.py [hunk 21]
```diff
@@ -153,6 +153,21 @@
...
 )
 
+# RSEI Aggregated Micro-data
+EPA_RSEI_NUMBER_FACILITIES_FIELD = "Number of facilities affecting the tract"
+EPA_RSEI_NUMBER_RELEASES_FIELD = "Number of releases affecting the tract"
+EPA_RSEI_NUMBER_CHEMICALS_FIELD = "Number of chemicals affecting the tract"
+EPA_RSEI_AVERAGE_TOXICITY_FIELD = (
+    "Average toxicity-weighted concentration of the cells in the tract"
+)
+EPA_RSEI_SCORE_FIELD = "RSEI Risk Score"
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/score/field_names.py:163]: 🎉

**Review by lucasmbrown-usds — CHANGES_REQUESTED** [review]
Looks good, but my comment about the headers from the CSV still stands from the last round of review. Thanks!

**Comment by saran-ahluwalia** [comment]
> Looks good, but my comment about the headers from the CSV still stands from the last round of review. Thanks!

My apologies. I overlooked that.

File: data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py (hunk 4 above)
Comment (saran-ahluwalia) [reply data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py:92]: Woah! I appreciate the fresh perspective @lucasmbrown-usds. I got so hung up on looking at the [data](https://github.com/usds/justice40-tool/issues/1070#issuecomment-1005604014)  that I didn't see the forest for the trees.

**Review by lucasmbrown-usds — APPROVED** [review]
LGTM! You've run this end-to-end, right? Thanks!!

**Pull request merged**